inference-engine/src/mkldnn_plugin/nodes/mkldnn_broadcast_node.cpp

```diff
@@ -0,0 +1,225 @@
+    }
+}
+
+void MKLDNNBroadcastNode::createPrimitive() {
+    for (int i = 0; i < getChildEdges().size(); i++) {
+        auto& dstMemPtr = getChildEdgeAt(i)->getMemoryPtr();
+        if (!dstMemPtr || !dstMemPtr->GetPrimitivePtr())
+            THROW_IE_EXCEPTION << "Destination memory " << i << "didn't allocate.";
+    }
+    for (int i = 0; i < getParentEdges().size(); i++) {
+        auto& srcMemPtr = getParentEdgeAt(i)->getMemoryPtr();
+        if (!srcMemPtr || !srcMemPtr->GetPrimitivePtr())
+            THROW_IE_EXCEPTION << "Input memory " << i << "didn't allocate.";
+    }
+    if (getSelectedPrimitiveDescriptor() == nullptr)
+        THROW_IE_EXCEPTION << "Preferable primitive descriptor is not set.";
+    if ((broadcast && getParentEdges().size() != 3) || (!broadcast && getParentEdges().size() != 2))
+        THROW_IE_EXCEPTION << "Incorrect number of input edges for layer " << getName();
+
+    if (optimizedCase) {
+        SizeVector srcBlockedDims = getParentEdgeAt(0)->getDesc().getBlockingDesc().getBlockDims();
+        SizeVector dstBlockedDims = getChildEdgeAt(0)->getDesc().getBlockingDesc().getBlockDims();
+        SizeVector srcBlockedStrides = getParentEdgeAt(0)->getDesc().getBlockingDesc().getStrides();
+        SizeVector dstBlockedStrides = getChildEdgeAt(0)->getDesc().getBlockingDesc().getStrides();
+
+        if (broadcastType == ngraph::op::AutoBroadcastType::EXPLICIT) {
+            SizeVector newSrcBlockedDims = SizeVector(dstBlockedDims.size(), 1);
+
+            for (int i = 0; i < getParentEdgeAt(2)->getDims()[0]; i++) {
+                newSrcBlockedDims[axesMapping[i]] = srcBlockedDims[i];
+            }
+
+            srcBlockedDims = newSrcBlockedDims;
+            srcBlockedStrides = updateStridesForDims(srcBlockedDims);
+        }
+
+        optimizedParams = prepareOptimizedParams(this, repeats, srcBlockedDims, dstBlockedDims, srcBlockedStrides, dstBlockedStrides);
+    }
+}
+
+void MKLDNNBroadcastNode::execute(mkldnn::stream strm) {
+    if (optimizedCase) {
+        optimizedExecute(this, optimizedParams, batchToProcess());
+    } else if (broadcast) {
+        ngraphExecute(this, broadcast);
+    } else {
+        // TODO: old behavior for IR's version < 10, remove when deprecated
+        auto &inputs = getSelectedPrimitiveDescriptor()->getConfig().inConfs;
+        auto &outputs = getSelectedPrimitiveDescriptor()->getConfig().outConfs;
+        size_t shape_size = inputs[1].desc.getDims()[0];
+        SizeVector dst_dims = outputs[0].desc.getDims();
+        SizeVector src_dims = inputs[0].desc.getDims();
+        SizeVector srcStrides = inputs[0].desc.getBlockingDesc().getStrides();
+        size_t data_size = inputs[0].desc.getPrecision().size();
+
+        if (!src_dims.size())
+            src_dims = SizeVector(1, 1);
+        if (!srcStrides.size())
+            srcStrides = SizeVector(1, 1);
+
+        if (dst_dims.size() != shape_size) {
```

Comment:
Seems it could be validated on node creation.

Reply:
It was a simple code transfer. Now it is validated earlier.